src/rebar_app_utils.erl

```diff
@@ -171,7 +171,7 @@ parse_deps(Parent, DepsDir, Deps, State, Locks, Level) ->
 %% (if it is a newer thing) or from the locks specified in the lockfile.
 -spec parse_dep(Dep, Parent, Dir, State, Locks, Level) -> rebar_app_info:t() when
       Dep :: tuple() | atom() | binary(), % TODO: meta to source() | lock()
-      Parent :: root | binary(),
+      Parent :: root | binary() | string(),
```

Comment:
I don't think we should be getting string names there? Where are they coming from?

Reply:
https://github.com/erlang/rebar3/blob/master/src/rebar_prv_plugins_upgrade.erl#L100
via
`-define(DEFAULT_PLUGINS_DIR, "plugins").`

I figured I'd prefer to fix the `-spec(_).` than the macro, but you tell me.

Reply:
I'd convert the default value to be the right type, but not change the macro. I.e. unicode:characters_to_binary(?DEFAULT_PLUGINS_DIR)

Reply:
dcaf5f6